### PR TITLE
MODREQMED-44 Fetch items from all tenants for "Send item in transit" 

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Purpose
Send item in transit is not working when the item is located in a different tenant from the current user’s affiliation

## Approach
Item should be fetched from any tenant

Resolves: [MODREQMED-44](https://folio-org.atlassian.net/browse/MODREQMED-44)

### CODE REVIEW

File: src/test/java/org/folio/mr/api/MediatedRequestActionsApiTest.java [hunk 1]
```diff
@@ -433,7 +453,18 @@ void sendItemInTransitFailsWhenMediatedRequestIsNotFoundByStatus() {
   }
 
   @SneakyThrows
-  private ResultActions sendItemInTransit(String itemBarcode) {
+  private ResultActions sendItemInTransit(String itemBarcode, MediatedRequestEntity request) {
+    wireMockServer.stubFor(WireMock.get(urlMatching("/search/instances" + ".*"))
```
Comment (OleksandrVidinieiev) [inline src/test/java/org/folio/mr/api/MediatedRequestActionsApiTest.java:457]: Does mocking search endpoint really have anything to do with sending item in transit? I'd keep these those two actions separate. Same for `MediatedRequestActionsApiTest#confirmItemArrival`.

File: src/test/java/org/folio/mr/service/MediatedRequestActionsServiceTest.java [hunk 2]
```diff
@@ -67,6 +72,12 @@ class MediatedRequestActionsServiceTest {
   @Mock
   private SearchService searchService;
 
+  @Mock
+  private SearchClient searchClient;
+
+  @Mock
+  private SystemUserScopedExecutionService executionService;
```
Comment (OleksandrVidinieiev) [inline src/test/java/org/folio/mr/service/MediatedRequestActionsServiceTest.java:79]: Here we mock `SystemUserScopedExecutionService` but never define any behavior for it. It is being called when you run your tests, right? What does it return?

File: src/test/java/org/folio/mr/service/MediatedRequestActionsServiceTest.java [hunk 3]
```diff
@@ -127,10 +140,12 @@ void sendItemInTransitSuccess() {
...
     when(mediatedRequestsRepository.save(any(MediatedRequestEntity.class)))
       .thenReturn(updatedRequest);
-    when(inventoryService.fetchItem(initialRequest.getItemId().toString()))
-      .thenReturn(new Item());
     when(mediatedRequestMapper.mapEntityToDto(any(MediatedRequestEntity.class)))
       .thenReturn(mappedRequest);
+    when(searchClient.searchInstance(anyString()))
+      .thenReturn(new SearchInstancesResponse()
+        .instances(List.of(new SearchInstance().items(List.of(
+          new SearchItem().id(initialRequest.getItemId().toString()))))));
```
Comment (OleksandrVidinieiev) [inline src/test/java/org/folio/mr/service/MediatedRequestActionsServiceTest.java:148]: Correct me if I'm wrong, but I believe the whole point of this fix is to be able to fetch an item from a different tenant. Do we actually do this in tests? If not, I really think we should.

**Review by alexanderkurash — APPROVED** [review]

File: src/main/java/org/folio/mr/service/impl/MediatedRequestActionsServiceImpl.java [hunk 4]
```diff
@@ -167,17 +172,39 @@ private MediatedRequestEntity updateMediatedRequestStatus(MediatedRequestEntity
...
-        .displaySummary(item.getDisplaySummary())
-        .copyNumber(item.getCopyNumber());
+    var searchInstances = searchClient.searchInstance(request.getInstanceId()).getInstances();
+    if (searchInstances == null || searchInstances.isEmpty()) {
+      log.info("extendMediatedRequest:: searchInstances not found");
+      return;
+    }
+    var searchInstance = searchInstances.get(0);
+    if (searchInstance == null || searchInstance.getItems() == null) {
+      log.info("extendMediatedRequest:: searchItems not found: {}",
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/mr/service/impl/MediatedRequestActionsServiceImpl.java:182]: warn?

**Review by OleksandrVidinieiev — APPROVED** [review]

**Pull request merged**